src/NuGet.Core/NuGet.Configuration/PackageSourceMapping/PackageSourceMappingProvider.cs

```diff
@@ -58,5 +58,39 @@ public void AddOrUpdatePackageSourceMappingSourceItem(PackageSourceMappingSource
 
             _settings.SaveToDisk();
         }
+
+        public void SavePackageSourceMappings(IReadOnlyList<PackageSourceMappingSourceItem> packageSourceMappingsSourceItems)
+        {
+            if (packageSourceMappingsSourceItems == null)
+            {
+                throw new ArgumentNullException(nameof(packageSourceMappingsSourceItems));
+            }
+
+            IReadOnlyList<PackageSourceMappingSourceItem> existingSettingsLookup = GetPackageSourceMappingItems();
+            if (existingSettingsLookup == null)
+            {
+                return;
```

Comment:
Does this mean `packageSourceMapping` should exist before this action?
Technically prevent from onboarding. Instead, should it create a new one?

Reply:
I removed this if statement because it was a bit redundant. GetPackageSourceMappingItems() will return an empty list of mappings if Package source mappings has not yet been enabled, so existingSettingsLookup will never be null. The Remove() and AddorUpdate() methods have logic that will create a new config if there are no previous mappings